src/Couchbase.Lite.Shared/API/Database/Database.cs

```diff
@@ -1265,40 +1250,62 @@ private void Save([NotNull]Document doc, C4Document** outDoc, C4Document* baseDo
+            if (resolvedDoc!=null && !ReferenceEquals(resolvedDoc, localDoc)) {
+                resolvedDoc.Database = this;
+            }
+
+            // The remote branch has to win, so that the doc revision history matches the server's.
             var winningRevID = remoteDoc.RevID;
             var losingRevID = localDoc.RevID;
 
             FLSliceResult mergedBody = (FLSliceResult)FLSlice.Null;
-            if (!ReferenceEquals(resolved, remoteDoc)) {
-                // Unless the remote revision is being used as-is, we need a new revision:
-                try {
-                    mergedBody = resolved.Encode();
-                } catch (ObjectDisposedException) {
-                    WriteLog.To.Sync.E(Tag, "Resolved document for {0} somehow got disposed!",
-                        new SecureLogString(resolved.Id, LogMessageSensitivity.PotentiallyInsecure));
-                    throw new RuntimeException(
-                        "Resolved document was disposed before conflict resolution completed.  Please file a bug report at https://github.com/couchbase/couchbase-lite-net");
+            C4RevisionFlags mergedFlags = 0;
+            if (!ReferenceEquals(resolvedDoc, remoteDoc)) {
+                var isDeleted = true;
+                if (resolvedDoc != null) {
+                    // Unless the remote revision is being used as-is, we need a new revision:
+                    try {
+                        mergedBody = resolvedDoc.Encode();
+                    } catch (ObjectDisposedException) {
+                        WriteLog.To.Sync.E(Tag, "Resolved document for {0} somehow got disposed!",
+                            new SecureLogString(resolvedDoc.Id, LogMessageSensitivity.PotentiallyInsecure));
+                        throw new RuntimeException(
+                            "Resolved document was disposed before conflict resolution completed.  Please file a bug report at https://github.com/couchbase/couchbase-lite-net");
+                    }
+                    if (mergedBody.Equals((FLSliceResult)FLSlice.Null))
+                        throw new RuntimeException("Resolved document contains a null body");
+                    isDeleted = resolvedDoc.IsDeleted;
+                } else if (resolvedDoc == null || resolvedDoc.IsEmpty) {
+                    FLEncoder* encoder = SharedEncoder;
+                    Native.FLEncoder_BeginDict(encoder, 0);
+                    Native.FLEncoder_EndDict(encoder);
+                    mergedBody = NativeRaw.FLEncoder_Finish(encoder, null);
+                    Native.FLEncoder_Reset(encoder);
                 }
+
+                if (isDeleted)
+                    mergedFlags |= C4RevisionFlags.Deleted;
             }
 
             // Tell LiteCore to do the resolution:
             C4Document* rawDoc = localDoc.c4Doc != null ? localDoc.c4Doc.RawDoc : null;
-            var flags = resolved.c4Doc != null ? resolved.c4Doc.RawDoc->selectedRev.flags : 0;
```

Comment:
This part worries me a bit because any other flags that happen to be in `resolved` are thrown away.  I think `mergedFlags` should be initialized to this line instead of just initialized to 0 above.